timemachine/cpp/src/tibd_exchange_move.cu

```diff
@@ -171,10 +169,42 @@ void TIBDExchangeMove<RealType>::move(
     curandErrchk(
         templateCurandUniform(this->cr_rng_samples_, this->d_sample_noise_.data, this->d_sample_noise_.length));
     k_generate_translations_inside_and_outside_sphere<<<1, d_rand_states_.length, 0, stream>>>(
-        this->steps_per_move_, d_box, d_center_.data, radius_, d_rand_states_.data, this->d_translations_.data);
+        this->num_proposals_per_move_, d_box, d_center_.data, radius_, d_rand_states_.data, this->d_translations_.data);
     gpuErrchk(cudaPeekAtLastError());
 
-    for (int step = 0; step < this->steps_per_move_; step++) {
+    /* --Algorithm Description--
+    * Targeted Insertion Biased Deletion algorithm is as follows
+    *
+    * 1. Generate all random noise upfront to ensure bitwise identical results regardless of batch size
+    * 2. Compute the initial weights of each of the molecules (no batching)
```

Comment:
Are the `initial_weights` recomputed at the start of each batch or the start of each `move()`?

Reply:
At the start of each `move()`